### ISSUE TITLE
Adding optimization presets

### ISSUE THREAD

**Issue by dmitriyyavorskiy** [excelsior-oss/excelsior-jet-maven-plugin#29]
It would be very important to have Optimization presets like in ExcelsiorJet GUI, classpath section. 

I have 50 jars in my project and want to compile just 5 of them, other set unchanged jars.

As is see we have 3 options there(Protect, Optimize and Pack in exe), it would be great to have package filter for example com.mpsdevelopment.* protect others not, or something like that.

**Comment by pjBooms** [comment]
Currently it works as follows:

All dependencies that share the same groupId as the pom.xml are compiled as a whole, all others are treated as libraries and compiled partially (only what is imported by others). It is roughly the same as SMART preset in the GUI. 

Please let us know, if you need more presets or have more flexible configuration, and if yes, please describe it in details.

**Comment by dmitriyyavorskiy** [comment]
I have some groupIds(current and others) 

com.mpsdevelopment.can - currect
com.mpsdevelopment.plasticine
com.mpsdevelopment.commons 

so currently i need to specify a list of groupIds or at least global groupId com.mpsdevelopment

**Comment by pjBooms** [comment]
So you actually need to compile more than now, right? 
What if I add the following configuration: 

```xml
<appGroupIds>
   <param> com.mpsdevelopment.can </param>
   <param> com.mpsdevelopment.plasticine </param>
   <param> com.mpsdevelopment.commons </param>
<appGroupIds>
```

with the default of pom.xml groupId. Will it be enough?

I can also check groupIds on startsWith, so it will be enough to set just
   `<param> com.mpsdevelopment </param>`

or use com.mpsdevelopment.* syntax for startsWith, so you will be able to compile com.mpsdevelopment.commons as a whole and not compile com.mpsdevelopment.commons.io.

Please let us know what do you prefer.

**Comment by dmitriyyavorskiy** [comment]
Yes, i think it is good for this project, wild cards are great, but maybe it is better to have the same config as for gui app
What do you think about such settings? 

<classpathEntries>
    <classpathEntry>
        <groupId>com.mpsdevelopment.can</groupId>
        <protect>all classes</protect>
        <optimize>all methods</optimize>
        <pack>auto-detect</pack>
    <classpathEntry>
    .... 
</classpathEntries>

and do not protect not specified entries

**Comment by pjBooms** [comment]
Ok, Multiple classpath entries can share the same groupId, so probably <classpathEntry> should also have <artifactId> configuration (and <version> as well). I will think about it.

**Comment by pjBooms** [comment]
Hello, the next plugin release will add dependencies management facility as you suggested. 
I have already written a draft of the plugin documentation of that future release: 

https://github.com/excelsior-oss/excelsior-jet-maven-plugin/tree/dependencies-management#dependencies-management

Please review it and tell if it meets your requirements.

**Comment by pjBooms** [comment]
The issue is fixed in 0.8.0 release
